### PR TITLE
feat(ctb): use HardforkOnlyProxy for predeploys

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Updates the migration to use HardforkOnlyProxy contracts for all predeployed contracts. ALL predeployed contracts are now behind HardforkOnlyProxy contracts that can only be mutated via hardfork.

### CODE REVIEW

File: op-bindings/predeploys/addresses.go [hunk 1]
```diff
@@ -39,10 +38,14 @@ var (
 	LegacyMessagePasserAddr           = common.HexToAddress(LegacyMessagePasser)
 	L2ERC721BridgeAddr                = common.HexToAddress(L2ERC721Bridge)
 	OptimismMintableERC721FactoryAddr = common.HexToAddress(OptimismMintableERC721Factory)
-	ProxyAdminAddr                    = common.HexToAddress(ProxyAdmin)
 	BaseFeeVaultAddr                  = common.HexToAddress(BaseFeeVault)
 	L1FeeVaultAddr                    = common.HexToAddress(L1FeeVault)
 
+	// HardforkOnlyProxyOwnerAddr is the address that owns all
```
Comment (smartcontracts) [inline op-bindings/predeploys/addresses.go:44]: Maybe not the right spot for this

File: op-chain-ops/genesis/check.go [hunk 2]
```diff
@@ -162,42 +156,68 @@ func PostCheckUntouchables(udb state.Database, currDB *state.StateDB, prevRoot c
...
-		// Sample storage slots to ensure that they are not modified.
-		var count int
+		// Iterate over all old storage slots.
+		var expCount int
 		expSlots := make(map[common.Hash]common.Hash)
 		err := prevDB.ForEachStorage(addr, func(key, value common.Hash) bool {
-			count++
+			expCount++
 			expSlots[key] = value
-			return count < MaxSlotChecks
```
Comment (smartcontracts) [inline op-chain-ops/genesis/check.go:187]: Remove MaxSlotChecks

File: op-chain-ops/genesis/check.go [hunk 3]
```diff
@@ -162,42 +156,68 @@ func PostCheckUntouchables(udb state.Database, currDB *state.StateDB, prevRoot c
...
-			return fmt.Errorf("error iterating over storage: %w", err)
+			return fmt.Errorf("error iterating over old storage: %w", err)
 		}
 
-		for expKey, expValue := range expSlots {
-			actValue := currDB.GetState(addr, expKey)
+		// Iterate over all new storage slots.
+		var actCount int
+		actSlots := make(map[common.Hash]common.Hash)
+		err = prevDB.ForEachStorage(addr, func(key, value common.Hash) bool {
```
Comment (smartcontracts) [inline op-chain-ops/genesis/check.go:184]: currDB

File: op-chain-ops/genesis/check.go [hunk 4]
```diff
@@ -162,42 +156,68 @@ func PostCheckUntouchables(udb state.Database, currDB *state.StateDB, prevRoot c
...
+			actCount++
+			actSlots[key] = value
+			return true
+		})
+		if err != nil {
+			return fmt.Errorf("error iterating over old storage: %w", err)
```
Comment (smartcontracts) [inline op-chain-ops/genesis/check.go:190]: new storage

File: op-chain-ops/genesis/check.go [hunk 5]
```diff
@@ -162,42 +156,68 @@ func PostCheckUntouchables(udb state.Database, currDB *state.StateDB, prevRoot c
...
+		}
+
+		// Assert that every old key still exists.
+		for expKey, expValue := range actSlots {
```
Comment (smartcontracts) [inline op-chain-ops/genesis/check.go:194]: expSlots

File: op-chain-ops/genesis/check.go [hunk 6]
```diff
@@ -309,7 +311,7 @@ func PostCheckPredeployStorage(db vm.StateDB, finalSystemOwner common.Address) e
 		for key, value := range expSlots {
 			// The owner slots for the L2XDM and ProxyAdmin are special cases.
```
Comment (smartcontracts) [inline op-chain-ops/genesis/check.go:312]: Fix comment

File: op-chain-ops/genesis/setters.go [hunk 7]
```diff
@@ -19,6 +19,7 @@ import (
 type ChainHashMap map[uint64]common.Hash
 
 var (
+
```
Comment (smartcontracts) [inline op-chain-ops/genesis/setters.go:22]: Unnecessary whitespace

**Review by mslipper — APPROVED** [review]
Ran this on a datadir and reviewed live with Kelvin. LGTM.
